### DISCUSSION TITLE
lettuce keeps reconnecting even when redis is unavailable

### DISCUSSION THREAD

**Discussion by devtravelport** [redis/lettuce#1659] [Q&A]
Hello All, I am trying to make lettuce stop reconnect tries when redis is down. I tried adding autoReconnect(false) and pingBeforeActivateConnection(true) even than with 30 ms connection timeout, lettuce taking 10 secs to try establish a connection and throw connect timeout. My guess is its still trying to reconnect. I dont have much logs to catch lettuce behavior. Any inputs will be helpful here.

**Comment by mp911de** [comment]
The point of auto-reconnect is to bridging Redis unavailability and retry to connect until Redis is available again. You can disable reconnects entirely. Alternatively, if you have knowledge over when your server will be available again, you can implement a custom Delay to delay reconnect attempts.

**Reply by devtravelport** [reply]
Thank you for your reply @mp911de . I am  not very familiar with lettuce, my understanding was - if autoReconnect is false , lettuce will not retry connection. How can I disable reconnect entirely? Could you suggest the option please.

**Reply by mp911de** [reply]
Exactly. Setting autoreconnect to false disables reconnects and command buffering when the connection is disconnected.

**Reply by devtravelport** [reply]
Okay I did the same but have enableAllAdaptiveRefreshTriggers also added to client. Does that cancels autoreconnect (false) ?

**Reply by mp911de** [reply]
`enableAllAdaptiveRefreshTriggers` is something entirely different, it controls when Lettuce attempts to refresh the topology. Check out the docs for further details (https://lettuce.io/core/release/reference/index.html#redis-cluster.refreshing-the-cluster-topology-view).

**Reply by devtravelport** [reply]
Hello Mark, Thank you for your reply. I tried just using autoReconnect flag as false and giving connection timeout. No other parameter. What I observe is when redis is down, irrespective of auto reconnect false, lettuce keeps trying to connect. I see a delay of 10 secs from when lettuce tried for connection and when it comes out of thread with exception as redis not reachable. I expected it to fail fast but it doesn't. Is there any default time setting on sync command to get connection - StatefulRedisClusterConnectionImpl.sync() ? Any input on this will be really helpful. I am facing 10 secs delay in case redis is down in my application for each redis call which overall increasing transaction time.

**Reply by mp911de** [reply]
Probably we're talking about two different things. Auto-reconnect applies to connections that were established once.

In your case, a node seems to be permanently down. Through the commands you want to invoke, Lettuce selects the unavailable node as target and every time a command is dispatched to the node, Lettuce attempts to create a connection. Since the faulty node is included in the topology, Lettuce attempts to use it.

I'd suggest also reducing the connect timeout. If 10 seconds are too much for your application, then reduce it to an acceptable value.
